src/timeseriesflattener/feature_cache/abstract_feature_cache.py

```diff
@@ -0,0 +1,41 @@
+"""Abstract method that defines a feature cache."""
```

Comment:
Files like these are often called `base_` instead of `abstract` but no strong feelings from me